### PR TITLE
doc: userguide: add documentation for flow aware scheduler mode

### PR DESCRIPTION
**Pull request by Bill-Fischofer-Linaro**
Update the ODP User Guide to include information on scheduler
capabilities and configuration and operating in flow aware mode.

Signed-off-by: Bill Fischofer <bill.fischofer@linaro.org>

### CODE REVIEW

**Comment by muvarov** [comment]
looks good for me. @lumag do you also want to review docs?

File: doc/users-guide/users-guide.adoc [hunk 1]
```diff
@@ -1184,11 +1184,179 @@ until the locking order for this lock for all prior events has been resolved
...
+* `odp_event_flow_id_set()`
+
+are used to query and set a 32-bit flow id associated with individual events.
+The assignment and interpretation of individual flow ids is under application
+control.
+
+When operating in flow aware mode, it is the combination of flow id and
+queue id that is used by the scheduler in making scheduling decisions. So,
+for example, an Atomic queue would normally be able to dispatch only a
+single event at a time. When operating in flow aware mode, however, the
```
Comment (psavol) [inline doc/users-guide/users-guide.adoc]: "... atomic queue would normally be able to dispatch events to a single thread at a time."

So, multiple events, but to a single thread.

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v2: 
* Wording changes clarifying atomic queue behavior as suggested by @psavol

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**